grails-app/jobs/org/pih/warehouse/jobs/UpdateExchangeRatesJob.groovy

```diff
@@ -6,6 +6,11 @@ class UpdateExchangeRatesJob {
 
     static concurrent = false
 
+    // By default this is true on QuartzDisplayJob, which invokes execute()
+    // and if sessionRequired is true, then QuartzDisplayJob tries to do session flush
+    // even if there is no session
+    static sessionRequired = false
+
```

Comment:
I haven't worked with jobs, hence I don't the flow, but I am wondering if when disabling this setting, we shouldn't still make sure that services that we are calling from `execute()` methods shouldn't be marked as @Transactional? 
If so: `currencyService`, `gparsService` are not @Transactional for example.

Reply:
Yeah, Transactional is still required (due to `allow_update_outside_transaction: false`). I'll add it in places that are missing it.